apps/src/codebridge/FileBrowser/index.tsx

```diff
@@ -212,11 +254,13 @@ export const FileBrowser = React.memo(() => {
           return;
         }
 
-        const existingFile = Object.values(project.files).some(
-          f => f.name === fileName && f.folderId === folderId
+        const duplicateNameError = getDuplicateFilenameError(
+          fileName,
+          folderId,
+          project.files
         );
-        if (existingFile) {
-          alert('File already exists');
+        if (duplicateNameError) {
+          alert(duplicateNameError);
           return;
         }
```

Comment:
It looks like we duplicate this logic 3 times (for new/move/rename). Would it be worth making a `checkForDuplicateFilename` to use in each place?

Reply:
I'll refactor this!